servlet/jakarta/build.gradle

```diff
@@ -104,41 +104,36 @@ dependencies {
 
 // Set up individual classpaths for each test, to avoid any mismatch,
 // and ensure they are only used when supported by the current jvm
-def undertowTest = tasks.register('undertowTest', Test) {
-    classpath = sourceSets.undertowTest.runtimeClasspath
-    testClassesDirs = sourceSets.undertowTest.output.classesDirs
-}
-def tomcat10Test = tasks.register('tomcat10Test', Test) {
-    classpath = sourceSets.tomcatTest.runtimeClasspath
-    testClassesDirs = sourceSets.tomcatTest.output.classesDirs
-
-    // Provide a temporary directory for tomcat to be deleted after test finishes
-    def tomcatTempDir = "$buildDir/tomcat_catalina_base"
-    systemProperty 'catalina.base', tomcatTempDir
-    doLast {
-        file(tomcatTempDir).deleteDir()
-    }
-
-    // tomcat-embed-core 10 presently performs illegal reflective access on
-    // java.io.ObjectStreamClass$Caches.localDescs and sun.rmi.transport.Target.ccl,
-    // see https://lists.apache.org/thread/s0xr7tk2kfkkxfjps9n7dhh4cypfdhyy
-    if (JavaVersion.current().isJava9Compatible()) {
-        jvmArgs += ['--add-opens=java.base/java.io=ALL-UNNAMED', '--add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED']
-    }
-}
-
-tasks.named("check").configure {
-    dependsOn undertowTest, tomcat10Test
-}
-
-// Only run these tests if java 11+ is being used
 if (JavaVersion.current().isJava11Compatible()) {
     def jetty11Test = tasks.register('jetty11Test', Test) {
         classpath = sourceSets.jettyTest.runtimeClasspath
         testClassesDirs = sourceSets.jettyTest.output.classesDirs
     }
 
+    def tomcat10Test = tasks.register('tomcat10Test', Test) {
+        classpath = sourceSets.tomcatTest.runtimeClasspath
+        testClassesDirs = sourceSets.tomcatTest.output.classesDirs
+
+        // Provide a temporary directory for tomcat to be deleted after test finishes
+        def tomcatTempDir = "$buildDir/tomcat_catalina_base"
+        systemProperty 'catalina.base', tomcatTempDir
+        doLast {
+            file(tomcatTempDir).deleteDir()
+        }
+    }
+    tasks.named('compileTomcatTestJava') { JavaCompile task ->
```

Comment:
`tasks.named(foo).configure {}` avoids the task from being created when unused.

Reply:
I believe named(string, action) also has the same semantics. From https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-org.gradle.api.Action-

> Locates a object by name, without triggering its creation or configuration, failing if there is no such object. The given configure action is executed against the object before it is returned from the provider.